src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
```

Comment:
I wouldn't hide enum values based on compile flags, that means any code that might want to use the enum value also has to have similar #ifdef guards.